controllers/utils/conditions.go

```diff
@@ -0,0 +1,21 @@
+package utils
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IsConditionTrue return true when the conditions contain a condition of given type and reason with status true
+func IsConditionTrue(conditions []v1.Condition, conditionType string, reason string) bool {
```

Comment:
I think the name is a bit confusing because it only refers to the status comparison.
Maybe IsSimilarConditionTrue ?

Reply:
`similar` can be anything...  but yes, will think about a better name

Reply:
>Actually I think it's fine as it is, functions in the meta package are similar.

Do note that `IsStatusConditionTrue` is invoking `IsStatusConditionPresentAndEqual` the later one naming convention is far better IMO.
I still think we can do a better job with the naming (even if `meta` didn't do so well), however I do consider this a Nit so anything you decide is fine with me.